pkg/internal/diskutil.go

```diff
@@ -324,13 +324,20 @@ func GetDeviceFSMap() (map[string]string, error) {
 func GetPVCreationLock(device string, symlinkDirs ...string) (ExclusiveFileLock, bool, []string, error) {
 	lock := ExclusiveFileLock{Path: device}
 	locked, err := lock.Lock()
-	if err != nil || !locked {
+	// If the device is busy, then we should continue and check for symlinks
+	if err != nil && err != unix.EBUSY {
 		return lock, false, []string{}, err
 	}
-	existingLinkPaths, err := GetMatchingSymlinksInDirs(device, symlinkDirs...)
-	if err != nil || len(existingLinkPaths) > 0 {
+	existingLinkPaths, symErr := GetMatchingSymlinksInDirs(device, symlinkDirs...)
+	// If symErr is not nil, there was an error fetching the symlinks
+	if symErr != nil {
```

Comment:
Rather than returning hardcoded `true` or `false` values - should we just just return `locked`  variable? There could be a case where device is locked but `GetMatchingSymlinksInDirs` erred out for different reason and in which case caller should not assume device did not get locked.

Reply:
This has been updated so that we're returning `locked` instead of true or false.